public/static/app/perf-bb/PerfBBRejectsCtrl.js

```diff
@@ -213,19 +243,17 @@ mciModule.controller('PerfBBRejectsCtrl', function (
         // NOTE do reload() here for server-side sorting
       });
 
-      api.core.on.filterChanged($scope, _.debounce(() => vm.reload(), 200));
-
-      api.selection.on.rowSelectionChanged(null, _.debounce(() => {
-        handleRowSelectionChange(api);
+      // handle row selection events
+      const rowSelectionChange = _.partial(handleRowSelectionChange,api);
+      api.selection.on.rowSelectionChanged(null, _.debounce((row) => {
+        rowSelectionChange(row);
         $scope.$apply();
       }));
+      api.selection.on.rowSelectionChangedBatch(null, rowSelectionChange);
 
-      // This is required when user selects all items
-      // (rowSelectionChanged doesn't work)
-      api.selection.on.rowSelectionChangedBatch(null, () => handleRowSelectionChange(api));
-
-      // Load initial set of data once `columns` are populated
-      api.core.on.rowsRendered(null, _.once(() => vm.reload()));
+      // api.core.on.filterChanged($scope, _.debounce(vm.reload, 200));
```

Comment:
[nit] Can we remove the commented out line?

Reply:
done